fizz.go

```diff
@@ -353,6 +353,20 @@ func InputModel(model interface{}) func(*openapi.OperationInfo) {
 	}
 }
 
+// Security adds a security definition to an operation.
+func Security(security *openapi.SecurityRequirement) func(*openapi.OperationInfo) {
+	return func(o *openapi.OperationInfo) {
+		o.Security = append(o.Security, security)
+	}
+}
+
+// NoSecurity explicitly removes all security.
+func NoSecurity() func(*openapi.OperationInfo) {
```

Comment:
Having this feels weird for the end-user. I'm wondering if it shouldn't be the default, applied directly by the package (i.e, if `operation.Security == nil`, initialize `Security` to empty slice, via `MarshalYAML/JSON`) ?

Reply:
Hmm... given that I understand your comment correctly I believe that would actually change semantics:
- you want to omit Security altogether (as per the spec) if you are not overriding the top level Security setting
- any serialized Security value, even an empty array has meaning.

From Operation Object:
> A declaration of which security mechanisms can be used for this operation. The list of values includes alternative security requirement objects that can be used. Only one of the security requirement objects need to be satisfied to authorize a request. To make security optional, an empty security requirement ({}) can be included in the array. This definition overrides any declared top-level security. **To remove a top-level security declaration, an empty array can be used.**

Reply:
Okay, that was me, I didn't read the spec properly. `NoSecurity()` would be useful only if the top-level `Security` field is set:

> To remove a top-level security declaration, an empty array can be used.

What would you think of `WithoutSecurity` instead as name ?

---

Speaking of that, the spec also specify the following:

>  To make security optional, an empty security requirement ({}) can be included in the array.

We could provide a "helper-func" (`WithOptionalSecurity()`) for that, that would append an empty `openapi.SecurityRequirement` object in the operation's `Security` field (slice). Thoughts ?

Reply:
I like it 👍